commands/outdated.js

```diff
@@ -11,13 +11,88 @@ type OutdatedArgs = {
   logger?: (...data: Array<mixed>) => void | mixed
 };
 type Outdated = (OutdatedArgs) => Promise<void>
+
+type Result = {
+  name: string,
+  range: Array<string>,
+  latest: string,
+};
 */
 
+/**
+ * Partitions the provided array into an array-of-arrays with inner arrays
+ * of size chunkSize or smaller.
+ */
+const partition /*: <T: any>(Array<T>, number) => Array<Array<T>> */ = (
+  arr,
+  chunkSize
+) => {
+  return [].concat.apply(
+    [],
+    arr.map(function (elem, i) {
+      return i % chunkSize ? [] : [arr.slice(i, i + chunkSize)];
+    })
+  );
+};
+
+/**
+ * An async implementation that mirrors Array.prototype.forEach.
+ */
+const forEachAsync /*: <TSrc: any, TDest: any>(Array<TSrc>, TSrc => Promise<TDest>) => Promise<void> */ = async (
+  arr,
+  callback
+) => {
+  await Promise.all(arr.map(callback));
+};
+
+/**
+ * Fetches metadata for the provided packages via the 'npm info' command.
+ */
+const fetchInfo = async (
+  packages /*: Array<string> */,
+  batchSize = 10
+) /*: Promise<{ [string]: {[string]: mixed} }> */ => {
+  let queries = {};
+  const setLatest = data => {
+    try {
+      const parsed /*: { [string]: string } */ = JSON.parse(data);
+      queries[parsed.name] = parsed;
+    } catch (e) {
+      /*do nothing*/
+    }
+  };
+
+  await forEachAsync(partition(packages, batchSize), async group => {
+    const flags = '-f version --json';
+    const cmd = `${node} ${yarn} npm info "${group.join(`" "`)}" ${flags}`;
+    try {
+      const result = await exec(cmd, {maxBuffer: 5 * 1024 * 1024});
+      result
+        .trim()
+        .split('\n')
+        .forEach(data => {
+          setLatest(data);
+        });
+    } catch (e) {
+      /* a single failure can cause the exec call to throw an exception.  In those
+       * cases, attempt to fetch for each package individually and only skip the
+       * failing package */
+      if (batchSize > 1) {
```

Comment:
Necessary to handle cases where a single package would fail the `npm info` call.  In that case, we can handle the remaining packages passed separately.

e.g.

```
jazelle yarn npm info "webworkify" "xhr" "package-does-not-exist" "fusion-core" -f version --json
```

will fail due to `package-does-not-exist` but we still want to fetch metadata for `webworkify`, `xhr`, and `fusion-core`.